backend/protocol_rpc/transactions_parser.py

```diff
@@ -0,0 +1,78 @@
+# rpc/transaction_utils.py
+
+from typing import Any
+import rlp
+import codecs
+from eth_account._utils.legacy_transactions import Transaction, vrs_from
+from eth_account._utils.signing import hash_of_signed_transaction
+from eth_account import Account
+from eth_utils import to_checksum_address
+from hexbytes import HexBytes
+from rlp.sedes import binary
+
+
+def decode_signed_transaction(raw_tx):
+    try:
+        txn_bytes = HexBytes(raw_tx)
+        tx_type = 0
+
+        tx = Transaction.from_bytes(txn_bytes)
+        msg_hash = hash_of_signed_transaction(tx)
+        vrs = vrs_from(tx)
+
+        # extracting sender address
+        sender = Account._recover_hash(msg_hash, vrs=vrs)
+        res = tx.as_dict()
+        # adding sender to result and cleaning
+
+        res["from"] = sender
+        res["to"] = to_checksum_address(f"0x{res['to'].hex()}") if res["to"] else None
+        res["data"] = res["data"].hex()
+        res["type"] = res.get("type", tx_type)
+
+    except Exception as e:
+        print("Error decoding transaction", e)
+        return None
+
+    return res
+
+
+def transaction_has_valid_signature(raw_tx: str, decoded_tx: dict) -> bool:
+    recovered_address = Account.recover_transaction(raw_tx)
+    # Compare the recovered address with the 'from' address in the transaction
+    return recovered_address == decoded_tx["from"]
+
+
+def decode_data_field(input: HexBytes) -> str:
```

Comment:
This is our custom way of decoding the data based on what we did in the frontend here: https://github.com/yeagerai/genlayer-simulator/pull/315/files#diff-86f9fcc5ab8dbb626a7814b69c5a32b5290572002ccc0405dcb608465faca384R185 
and here: https://github.com/yeagerai/genlayer-simulator/pull/315/files#diff-86f9fcc5ab8dbb626a7814b69c5a32b5290572002ccc0405dcb608465faca384R283

Reply:
@cristiam86 Is this documented somewhere? seems like something we might document about "how transactions are signed"

Reply:
https://github.com/orgs/yeagerai/projects/2/views/2?filterQuery=-status%3Adone+docs&pane=issue&itemId=74589262